src/parachain/refund.ts

```diff
@@ -72,44 +61,30 @@ export class DefaultRefundAPI extends DefaultTransactionAPI implements RefundAPI
     async list(): Promise<RefundRequestExt[]> {
         const head = await this.api.rpc.chain.getFinalizedHead();
         const refundRequests = await this.api.query.refund.refundRequests.entriesAt(head);
-        return refundRequests.map((v) => parseRefundRequest(v[1], this.btcNetwork, this.wrappedCurrency));
-    }
-
-    async mapForUser(account: AccountId): Promise<Map<H256, RefundRequestExt>> {
-        const refundPairs: [H256, RefundRequest][] = await this.api.rpc.refund.getRefundRequests(account);
-        const mapForUser: Map<H256, RefundRequestExt> = new Map<H256, RefundRequestExt>();
-        refundPairs.forEach((refundPair) =>
-            mapForUser.set(refundPair[0], parseRefundRequest(refundPair[1], this.btcNetwork, this.wrappedCurrency))
-        );
-        return mapForUser;
+        return refundRequests
+            .filter(([_, req]) => req.isSome.valueOf)
+            .map((v) => parseRefundRequest(v[1].unwrap(), this.btcNetwork, this.wrappedCurrency));
     }
 
     async getRequestById(refundId: H256): Promise<RefundRequestExt> {
         const head = await this.api.rpc.chain.getFinalizedHead();
-        return parseRefundRequest(
-            await this.api.query.refund.refundRequests.at(head, refundId),
-            this.btcNetwork,
-            this.wrappedCurrency
-        );
-    }
-
-    async getRequestIdByIssueId(issueId: H256 | string): Promise<H256> {
         try {
-            const id = ensureHashEncoded(this.api, issueId);
-            const keyValuePair = await this.api.rpc.refund.getRefundRequestsByIssueId(id);
-            return keyValuePair[0];
+            return parseRefundRequest(
+                (await this.api.query.refund.refundRequests.at(head, refundId)).unwrap(),
+                this.btcNetwork,
+                this.wrappedCurrency
+            );
         } catch (error) {
             return Promise.reject(new Error(`Error fetching refund request by issue id: ${error}`));
         }
     }
 
-    async getRequestByIssueId(issueId: H256 | string): Promise<RefundRequestExt> {
-        try {
-            const id = ensureHashEncoded(this.api, issueId);
-            const keyValuePair = await this.api.rpc.refund.getRefundRequestsByIssueId(id);
-            return parseRefundRequest(keyValuePair[1], this.btcNetwork, this.wrappedCurrency);
-        } catch (error) {
-            return Promise.reject(new Error(`Error fetching refund request by issue id: ${error}`));
+    async getRequestByIssueId(issueId: string): Promise<RefundRequestExt> {
+        const list = await this.list();
+        const requests = list.filter((request) => request.issueId === issueId);
+        if (requests.length !== 1) {
+            return Promise.reject(new Error(`No refund request fund for issue ID ${issueId}`));
         }
+        return requests[0];
```

Comment:
I'm a little confused by this - we're filtering the list, which could return multiple entries, then throwing an error if the result is longer than 1. What are the circumstances where we could get more than one result? e.g. could there be multiple matches for an issueId? If this is possible, then I would expect to throw two errors: one for no match, and one for multiple matches. If there can only ever be a single match, then I'd prefer to use `list.find`, as it makes the code a bit clearer (i.e. by using `find`, it's implicit that we're looking for a single item; by using `filter` the implication is that we're looking for 1 or more items).

I am looking at this function in isolation though, so there could well be some context I'm missing :)

Reply:
Good advice, thanks!